### PR TITLE
Day 17: Search for an integer in a matrix

### PR DESCRIPTION
**Pull request by alberthu16**
Please comment, critique, and discuss! :D

### CODE REVIEW

File: day17/searchMatrix.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    assert not searchMatrix([[1],[2],[3],[4],[5]], 6)
+
+    t = [[1, 2, 3],
+         [4, 5, 6],
+         [7, 8, 9]]
+    assert searchMatrix(t, 1)
+    assert searchMatrix(t, 2)
+    assert searchMatrix(t, 5)
+    assert searchMatrix(t, 9)
+
```
Comment (apengwin) [inline day17/searchMatrix.py]: I think there's  an `O(m + n)` solution to this. :)

File: day17/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+Question of the day: https://leetcode.com/problems/search-a-2d-matrix/#/description
+
+Write an efficient algorithm that searches for a value
+in an `n` x `m` matrix. This matrix has the following
+properties:
+
+- Integers in each row are sorted from left to right.
+- The first integer of each row is greater than the last
+integer of the previous row.
+
```
Comment (apengwin) [inline day17/README.md]: I think there's an interesting variation to this problem when the matrix is a [Young tableau](https://en.wikipedia.org/wiki/Young_tableau)

File: day17/searchMatrix.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    # first and last elements edge case
+    endCoord = indexToCoord(lowerBound, numCols)
+    if matrix[endCoord[0]][endCoord[1]] == value:
+        return True
+
+    return False
+
+def indexToCoord(index, matrixWidth):
+    return (index/matrixWidth, index%matrixWidth)
```
Comment (jjojohjohn) [inline day17/searchMatrix.py]: i agree with that guy

File: day17/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+`4 % len(matrix) == 0` and `4 / len(matrix) == 1`
+
+so our column index is 0 and our row index is 1.
+
+Now that we have a translation mechanism, we can just do binary
+search on the matrix, treating the coordinates like the indices
+of a 1D array.
+
+This solution is `O(logn)` runtime and `O(1)` space since we no
```
Comment (jjojohjohn) [inline day17/README.md]: log(m*n) not log(n) i believe?

File: day17/searchMatrix.py (hunk 3 above)
Comment (alberthu16) [inline day17/searchMatrix.py]: u right

File: day17/README.md (hunk 4 above)
Comment (alberthu16) [inline day17/README.md]: u right, i fell into the trap of blanket-ing an input size as `n`

**Pull request merged**